src/mca/rmaps/base/rmaps_base_frame.c

```diff
@@ -327,15 +327,20 @@ static int check_modifiers(char *ck, prte_job_t *jdata,
                                 "mapping policy", ck2[i]);
                 return PRTE_ERR_SILENT;
             }
-            if (NULL == (ptr = strchr(ck2[i], '='))) {
-                /* missing the value */
+            /* Should be processing PE=<n>, verify '=' is present then that following value
```

Comment:
Should we be doing the same thing for every option that requires an '='? Seems to me that "PE-LIST", for one, suffers the same potential issue - yes?

Reply:
I think better parsing of options where we expect '=' is a good idea, and probably validating numerics after the '=', I can handle that if you want. I did think about checking for 'PE=' instead of 'PE' in the preceding strcasencmp but decided not to since it didn't seem to follow the code pattern.

Reply:
That would be nice, if you wouldn't mind. I have no issue with including the '=' in the strcasecmp - seems like that would make life a bit easier.

Reply:
I fixed processing for the :pe= and :pe-list qualifiers. I also added a new error message since the existing one wasn't quite right.

I noticed that invalid values are accepted as numerics for the --map-by ppr:<n>:<resource> and also if the ppr qualifier has :pe= or :pe-list= as part of it's sting. I asked @jjhursey about fixing this since I thought it was outside the scope of this pull request and he agreed it should be deferred and fixed in another pull request later.